member/src/main/scala/com/improving/app/member/domain/MemberValidation.scala

```diff
@@ -92,10 +211,99 @@ object MemberValidation {
       validateLastName(memberInfo.lastName),
       memberInfo.notificationPreference.validNec,
       memberInfo.notificationOptIn.validNec,
-      validateContact(memberInfo),
-      validateOrganizations(memberInfo),
-      validateTenant(memberInfo),
+      validateRequiredContact(memberInfo),
+      memberInfo.organizationMembership.validNec,
+      validateRequiredTenant(memberInfo),
       memberInfo.unknownFields.validNec
     ).mapN(MemberInfo.apply)
 
-}
+  def validateEditableInfo(
+    editableInfo: EditableInfo
+  ): ValidationResult[EditableInfo] =
+    (
+      editableInfo.handle.validNec,
+      editableInfo.avatarUrl.validNec,
+      validateFirstNameOpt(editableInfo.firstName),
+      validateLastNameOpt(editableInfo.lastName),
+      editableInfo.notificationPreference.validNec,
+      validateContactOpt(editableInfo.contact, editableInfo.notificationPreference),
+      validateOrganizationMembership(editableInfo.organizationMembership),
+      validateTenantOpt(editableInfo.tenant),
+      editableInfo.unknownFields.validNec
+    ).mapN(EditableInfo.apply)
+
+  def validateMemberInfoOpt(
+    infoOpt: Option[MemberInfo]
+  ): ValidationResult[Option[MemberInfo]] =
+    infoOpt.fold[ ValidationResult[Option[MemberInfo]]](NoMemberInfoAssociated.invalidNec) {
+      validateMemberInfo(_).map(Some(_))
+    }
+
+  def validateEditableInfoOpt(
+    infoOpt: Option[EditableInfo]
+  ): ValidationResult[Option[EditableInfo]] =
+    infoOpt.fold[ValidationResult[Option[EditableInfo]]](NoEditableInfoAssociated.invalidNec) {
+      validateEditableInfo(_).map(Some(_))
+    }
+
+  def validateRegisterMember(
+    registerMember: RegisterMember
+  ): ValidationResult[RegisterMember] = (
+    (
+      validateMemberIdOpt(registerMember.memberId),
+      validateMemberInfoOpt(registerMember.memberInfo),
+      validateMemberIdOpt(registerMember.registeringMember),
+      registerMember.unknownFields.validNec
+    ).mapN(RegisterMember.apply)
```

Comment:
What is this doing?

Reply:
Refer to this for a more detailed view https://typelevel.org/cats/datatypes/validated

Simply put, this function validates all the fields of whatever is in that MapN function.

Shailendra, the previous writer of Member, decided to use this as the validation. I continued to use this as this provided the infrastructure to complete creating Member without changing too much that may be out of scope, so that's why I kept it for now.